{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "c0175e59-8718-4a95-9398-cfe378ef131e",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " * Serving Flask app '__main__'\n",
      " * Debug mode: off\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.\n",
      " * Running on http://127.0.0.1:5000\n",
      "Press CTRL+C to quit\n",
      "127.0.0.1 - - [06/Jan/2023 19:33:34] \"GET /chatbot?pergunta=em%20que%20foi%20o%20filme%20interestelar%20foi%20lançado%3F HTTP/1.1\" 200 -\n",
      "127.0.0.1 - - [06/Jan/2023 19:34:24] \"GET /chatbot?pergunta=img%3A%20você%20pode%20desenhar%20a%20nave%20do%20filme%20interestelar%3F HTTP/1.1\" 200 -\n",
      "[2023-01-06 19:35:19,743] ERROR in app: Exception on /chatbot [GET]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\flask\\app.py\", line 2525, in wsgi_app\n",
      "    response = self.full_dispatch_request()\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\flask\\app.py\", line 1822, in full_dispatch_request\n",
      "    rv = self.handle_user_exception(e)\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\flask_cors\\extension.py\", line 165, in wrapped_function\n",
      "    return cors_after_request(app.make_response(f(*args, **kwargs)))\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\flask\\app.py\", line 1820, in full_dispatch_request\n",
      "    rv = self.dispatch_request()\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\flask\\app.py\", line 1796, in dispatch_request\n",
      "    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)\n",
      "  File \"C:\\Users\\webma\\AppData\\Local\\Temp/ipykernel_7772/922946707.py\", line 32, in chatbot\n",
      "    resposta = mydraw(pergunta)\n",
      "  File \"C:\\Users\\webma\\AppData\\Local\\Temp/ipykernel_7772/922946707.py\", line 19, in mydraw\n",
      "    response = openai.Image.create(\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\openai\\api_resources\\image.py\", line 22, in create\n",
      "    return instance.request(\"post\", cls._get_url(\"generations\"), params)\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\openai\\openai_object.py\", line 179, in request\n",
      "    response, stream, api_key = requestor.request(\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\openai\\api_requestor.py\", line 181, in request\n",
      "    resp, got_stream = self._interpret_response(result, stream)\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\openai\\api_requestor.py\", line 396, in _interpret_response\n",
      "    self._interpret_response_line(\n",
      "  File \"C:\\Users\\webma\\anaconda3\\lib\\site-packages\\openai\\api_requestor.py\", line 429, in _interpret_response_line\n",
      "    raise self.handle_error_response(\n",
      "openai.error.InvalidRequestError: Your request was rejected as a result of our safety system. Your prompt may contain text that is not allowed by our safety system.\n",
      "127.0.0.1 - - [06/Jan/2023 19:35:19] \"GET /chatbot?pergunta=img%3A%20você%20poderia%20criar%20uma%20imagem%20realista%20de%20um%20buraco%20negro%3F HTTP/1.1\" 500 -\n",
      "127.0.0.1 - - [06/Jan/2023 19:36:40] \"GET /chatbot?pergunta=img%3A%20gere%20um%20carro%20de%20formula%201%20realista%20e%20em%203d HTTP/1.1\" 200 -\n",
      "127.0.0.1 - - [06/Jan/2023 19:37:08] \"GET /chatbot?pergunta=Quem%20foi%20Carl%20Sagan%3F HTTP/1.1\" 200 -\n"
     ]
    }
   ],
   "source": [
    "from flask import Flask, render_template, request, jsonify\n",
    "from flask_cors import CORS\n",
    "import openai\n",
    "\n",
    "app = Flask(__name__)\n",
    "CORS(app)\n",
    "\n",
    "openai.api_key = \"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\"\n",
    "\n",
    "def mychat(prompt):\n",
    "    response = openai.Completion.create(\n",
    "            model = \"text-davinci-003\",\n",
    "            prompt=prompt,\n",
    "            max_tokens = 2048\n",
    "    )\n",
    "    return response.choices[0].text\n",
    "\n",
    "def mydraw(prompt):\n",
    "    response = openai.Image.create(\n",
    "        size = \"512x512\",\n",
    "        prompt = prompt,\n",
    "        n = 1,\n",
    "        response_format = \"url\"\n",
    "    )\n",
    "    return response[\"data\"][0][\"url\"]\n",
    "\n",
    "@app.route('/chatbot')  \n",
    "def chatbot():\n",
    "        pergunta = request.args.get('pergunta')\n",
    "        if pergunta[0:4] == 'img:' :\n",
    "            pergunta = pergunta.replace('img:','')\n",
    "            resposta = mydraw(pergunta)\n",
    "            return jsonify(url_imagem=resposta, resposta ='')\n",
    "        else:\n",
    "            resposta = mychat(pergunta)\n",
    "            return jsonify(resposta=resposta)\n",
    "\n",
    "app.run()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a915d28-b1d9-4221-82bd-63fba62df498",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
